persistence-api/src/main/java/io/stargate/db/EventListenerWithChannelFilter.java

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db;
+
+import java.net.InetAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Specialized version of EventListener which accepts a filter of which channels to send the event
+ * to.
+ */
+public interface EventListenerWithChannelFilter extends EventListener {
+  void onCreateKeyspace(String keyspace, Predicate<Map<String, String>> headerFilter);
+
+  @Override
+  default void onCreateKeyspace(String keyspace) {
+    onCreateKeyspace(keyspace, c -> true);
```

Comment:
Using `null` instead of `c -> true` will allow the channel matcher to early exit without calling `toHeaders()` and the predicate for the default implementation.